pkg/operator/ceph/object/user/controller.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package objectuser to manage a rook object store user.
+package objectuser
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coreos/pkg/capnslog"
+	opkit "github.com/rook/operator-kit"
+	cephv1beta1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1beta1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephrgw "github.com/rook/rook/pkg/daemon/ceph/rgw"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	customResourceName       = "objectstoreuser"
+	customResourceNamePlural = "objectstoreusers"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "op-object")
+
+// ObjectStoreResource represents the object store user custom resource
+var ObjectStoreUserResource = opkit.CustomResource{
+	Name:    customResourceName,
+	Plural:  customResourceNamePlural,
+	Group:   cephv1beta1.CustomResourceGroup,
+	Version: cephv1beta1.Version,
+	Scope:   apiextensionsv1beta1.NamespaceScoped,
+	Kind:    reflect.TypeOf(cephv1beta1.ObjectStoreUser{}).Name(),
+}
+
+// ObjectStoreUserController represents a controller object for object store user custom resources
+type ObjectStoreUserController struct {
+	context     *clusterd.Context
+	rookImage   string
```

Comment:
you shouldn't need the `rookImage` or the `hostNetwork` properties for this controller